### PR TITLE
Fix eDP flicker

### PR DESCRIPTION
**Pull request by crawfxrd**
Cherry-pick the commits needed to fix eDP flicker on oryp7.

### CODE REVIEW

**Comment by leviport** [comment]
I built this from source since the build server takes a while, and after about a dozen manual suspends & resumes, I'm not seeing the screen freak out once. It used to happen basically every resume on this unit. It still needs more testing before approval, but this ought to do the trick!

**Review by jackpot51 — APPROVED** [review]

**Review by bflanagin — APPROVED** [review]
Flickering Fixed.

**Pull request merged**
